.github/workflows/pr.yaml

```diff
@@ -0,0 +1,43 @@
+name: pr
+on: [ pull_request ]
+permissions:
+  contents: read
+jobs:
+  semgrep:
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    runs-on: ubuntu-latest
+    container:
+      image: returntocorp/semgrep
+    steps:
+      # Retrieve the source code for the repository
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+      # Fetch the semgrep rules
+      - run: git clone https://github.com/dgryski/semgrep-go.git
+      # Run the rule checker using the fetched rules
+      - run: semgrep ci -f semgrep-go
+
+  check-race:
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
why do you propose to test two times: with and without `-race`?

Reply:
Race takes significant longer and that's why it makes sense to break out.